contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,200 @@
+ 
+2. **Angular uses TypeScript.** Angular applications are built using TypeScript language, a superset of JavaScript. TypeScript provides static typing. This helps the compiler show warnings about any potential errors in the code, before the app is even run. Another advantage of TypeScript is that it provides code completion using IntelliSense. IntelliSense provides active hints as code is added.
+
+    <box> TypeScript was created and is maintained by Microsoft, and has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> for the last few years making it reliable to use.</box>
+                                                  
+3. **Google Long-Term Support.** Some software engineers consider the fact that Angular is supported by Google an advantage of the technology. While this may sound justified, it does not serve as much of an incentive. The good sign though is that Google announced Long-Term Support (LTS) for the technology back in 2017.
+
+    [source](https://hackr.io/blog/why-should-you-learn-angular-in-2018)
+
+### Benefit: Angular provides a declarative user interface.
+
+Angular uses HTML to define the app’s user interface. HTML is a declarative language which is more intuitive than using imperative code to define the UI. An Imperative approach requires you to tell the compiler how you want the program to run by focussing on the logic implementation. Declarative code, on the other hand enables you to describe the logic without worrying about the implementation. How does it help? You do not need to invest time in dealing with how your program runs and deciding what to load first.
+
+Lets see a working example. 
+
+With jQuery you might write the following code to define a textbox that takes numeric values and doesn’t allow alphanumeric input or input that is beyond of a defined range.
+
+```
+$("#textbox").numericTextBox({
+    value: 10,
+    min: -10,
+    max: 100,
+    step: 0.75,
+    format: "n",
+    decimals: 3
+});
+```
+On the other hand, using versions Angular 2 and above, your code would look something like this:
+
+```
+<input numerictextbox k-min="-10" k-max="100" k-step="0.75"
+  k-format="n" k-decimals="3"/>
+```
+Instead of worrying about how to transform the element, you can declare what you want the user to see.
+Rather than spending time thinking on how the program should flow, you can define what you want and Angular will take care of the dependencies.
+
+[source](https://www.telerik.com/blogs/three-ds-of-web-development-1-declarative-vs-imperative)
+
+### Benefit: Simplified MVC pattern and reduced coding.
+
+Angular framework is embedded with the MVC (Model-View-Controller) software architectural setup. The HTML, is what is presented to the user (along with some CSS for layout). This represents the View component. Next to it, we have the `component.ts` files. This is the controller. Essentially, it can choose which data to push to our view (`.html`). 
```

Comment:
> embedded with the MVC (Model-View-Controller) software architectural setup

Maybe rephrase to something like `use the MVC model`. 

~~A link describing more details would be good~~. 
**EDIT**: I saw it below after the diagram. Maybe the paragraph with the link should come first? Not sure if show-before-tell is applicable in this case as well, @damithc your thoughts?

Also, instead of describing it in lengthy text, showing an example of the folder structure would be more intuitive IMO.